src/libexpr/parser.y

```diff
@@ -31,8 +31,25 @@
 #define YY_DECL int yylex \
     (YYSTYPE * yylval_param, YYLTYPE * yylloc_param, yyscan_t yyscanner, nix::ParserState * state)
 
+// For efficiency, we only track offsets; not line,column coordinates
+# define YYLLOC_DEFAULT(Current, Rhs, N)                                \
+    do                                                                  \
+      if (N)                                                            \
+        {                                                               \
+          (Current).beginOffset = YYRHSLOC (Rhs, 1).beginOffset;        \
+          (Current).endOffset  = YYRHSLOC (Rhs, N).endOffset;           \
+        }                                                               \
+      else                                                              \
+        {                                                               \
+          (Current).beginOffset = (Current).endOffset =                 \
+            YYRHSLOC (Rhs, 0).endOffset;                                \
+        }                                                               \
+    while (0)
+
 namespace nix {
 
+typedef std::map<PosIdx, DocComment> DocCommentMap;
```

Comment:
```suggestion
typedef std::unordered_map<PosIdx, DocComment> DocCommentMap;
```

Reply:
Almost! I've prepared this change, but had to revert for now https://github.com/NixOS/nix/pull/11072/commits/6a125e65d0f2dac90cdc69f16be0a4bd888c7a2f